packages/scene-composer/src/components/panels/AddComponentMenu.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import { defineMessages, MessageDescriptor, useIntl } from 'react-intl';
+import * as THREE from 'three';
+import { IconProps } from '@awsui/components-react';
+
+import { ToolbarItemOptionRaw, ToolbarItemOptions } from '../toolbars/common/types';
+import { sceneComposerIdContext } from '../../common/sceneComposerIdContext';
+import { IDataOverlayComponentInternal, useStore } from '../../store';
+import { findComponentByType } from '../../utils/nodeUtils';
+import { COMPOSER_FEATURES, KnownComponentType } from '../../interfaces';
+import { ToolbarItem } from '../toolbars/common/ToolbarItem';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { Component } from '../../models/SceneModels';
+import { IDataBindingComponentInternal } from '../../store/internalInterfaces';
+
+interface AddComponentMenuProps {
+  onSelect?: (selectedObject: ObjectTypes) => void;
+}
+
+// Note: ObjectType String is used to record metric. DO NOT change existing ids unless it's necessary.
+enum ObjectTypes {
+  Component = 'add-component',
+  Overlay = 'add-component-overlay',
+  DataBinding = 'add-component-data-binding',
+}
+
+type AddComponentMenuItem = ToolbarItemOptions & {
+  uuid: ObjectTypes;
+};
+
+const labelStrings: { [key in ObjectTypes]: MessageDescriptor } = defineMessages({
```

Comment:
Nit: What are you expecting to gain from this type annotation?

Reply:
I want to make sure all types in ObjectTypes will get a label string. with this type set, there will be error if new types are not added to this strings map